### PR TITLE
docs: add release notes for 0.17.10

### PR DESCRIPTION
**Pull request by mackrorysd**
Add release notes for 0.17.10.

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -8,6 +8,34 @@
...
+**Release Date:** March 03, 2022
+
+**Breaking Change**
+
+-  API: Pytorch Lightning has been updated from 1.3.5 to 1.5.9 to address a security vulnerability.
+   Experiments using Pytorch Lightning Adapter with v1.3.5 are no longer supported.
+
+**New Features**
+
+-  Added self-supervised image classification example in PyTorch using Bootstrap Your Own Latent
```
[the remark this answers is not in the document]
Comment (mackrorysd) [reply docs/release-notes.txt:23]: There isn't (https://determinedai.atlassian.net/browse/DET-6450). @coreystatendet could you elaborate on the original release note, perhaps?

File: docs/release-notes.txt [hunk 2]
```diff
@@ -8,6 +8,34 @@
...
+   (BYOL).
+
+-  PyTorchTrial and TFKerasTrial now automatically log the number of batches and number of records
```
[the remark this answers is not in the document]
Comment (mackrorysd) [reply docs/release-notes.txt:26]: I would imagine it's helpful as a sort of progress-bar type indicator. @rb-determined-ai anything you'd care to add to the original text here, given Gene's question?

File: docs/release-notes.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mackrorysd) [reply docs/release-notes.txt]: Seems to me the "benefit" is self-supervised learning, so how about we just emphasize more. @gh-determined-ai what do you think of Added PyTorch example using Bootstrap Your Own Latent to do self-supervised (no labels) image classification"

File: docs/release-notes.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply docs/release-notes.txt:25]: Yes, it's hard for the user to gather this information for themself directly.  Separately, we are working on adding a few callbacks what would make measuring it more possible, but this is a generally nice thing to give to users, and it's an easy feature to give to them out-of-the-box.  So yeah, a lot like what @mackrorysd is saying.

[the remark this answers is not in the document]
Comment (mackrorysd) [reply docs/release-notes.txt:25]: How about "PyTorchTrial and TFKerasTrial now automatically log the number of batches and number of records to make tracking progress easier"?

File: docs/release-notes.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/release-notes.txt]: Looks good. Minor changes:
Added PyTorch example using Bootstrap Your Own Latent (BYOL) to do self-supervised, no labels, image classification.

File: docs/release-notes.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (gh-determined-ai) [reply docs/release-notes.txt:25]: Good.

File: docs/release-notes.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mackrorysd) [reply docs/release-notes.txt]: :+1:

**Pull request merged**